web3/module.py

```diff
@@ -40,6 +43,8 @@ def retrieve_blocking_method_call_fn(
 ) -> Callable[..., RPCResponse]:
     def caller(*args: Any, **kwargs: Any) -> RPCResponse:
         (method_str, params), response_formatters = method.process_params(module, *args, **kwargs)
+        if method_str == 'existing_filter':
```

Comment:
First thought is that we can at least get rid of the polymorphic return type for the `method_str` by converting this to use an exception instead of a return value.

```python
try:
    (method_str, params), response_formatters = method.process_params(...)
except _UseExistingFilterID as err:
    return LogFilter(module=module, filter_id=err.filter_id)
...
```

Reply:
Sorta ugly but at least the mechanism doesn't mix-up the return types.....

Reply:
Ooh, I like that better 👍